src/Uno.Toolkit.UITest/Behaviors/FlipViewPipsPagerBehavior/Given_PipsPagerExtensions.cs

```diff
@@ -0,0 +1,71 @@
+﻿using NUnit.Framework;
+using Uno.Toolkit.UITest.Extensions;
+using Uno.UITest.Helpers.Queries;
+
+namespace Uno.Toolkit.UITest.Behaviors.FlipViewPipsPagerBehavior;
+
+public class Given_PipsPagerExtensions : TestBase
+{
+	protected override string SampleName => "FlipViewExtensions";
+
+	[Test]
+	public void Number_Of_Items_Should_Be_The_Same()
+	{
+		var (flipViewCounter, pipsPager) = GetPipsPagerAndLabelElements();
+
+		var pipsPagerItemsCount = pipsPager.GetDependencyPropertyValue<int>("NumberOfPages");
+		var flipViewCount = flipViewCounter.GetDependencyPropertyValue<string>("Text");
+
+		Assert.AreEqual(pipsPagerItemsCount, int.Parse(flipViewCount));
+	}
+
+	[Test]
+	public void Number_Of_Items_Should_Be_The_Same_AfterAddingMoreItems()
+	{
+		var (flipViewCounter, pipsPager) = GetPipsPagerAndLabelElements();
+
+		App.Tap("AddNewPageButton");
+
+		var pipsPagerItemsCount = pipsPager.GetDependencyPropertyValue<int>("NumberOfPages");
+		var flipViewCount = flipViewCounter.GetDependencyPropertyValue<string>("Text");
+
+		Assert.AreEqual(pipsPagerItemsCount, int.Parse(flipViewCount));
```

Comment:
This test fails to validate if the count has actually increased after triggering `AddNewPageButton`.
You should check the before AND after for both `NumberOfPages` + `Text`.